### PR TITLE
test(solana): extend e2e test coverage for unhappy paths in GMP app

### PR DESCRIPTION
**Pull request by mariuszzak**
## Description

Added new e2e tests for GMP app.

  1. `Test_GMPTimeoutFromCosmos` - Tests GMP packet timeout when sent from Cosmos to Solana
  2. `Test_GMPFailedExecutionFromCosmos` - Tests CPI execution failure handling (Cosmos -> Solana)
  3. `Test_GMPFailedExecutionFromSolana` - Tests error acknowledgment flow (Solana -> Cosmos)

Additionally:
1. Added helper functions for PDA calculation (`pda.go`).
2. Reorganized the test helpers into corresponding modules. 
3. Extended documentation for each GMP e2e test.
4. Removed `create_legacy_tx` from relayer.

ref: IBCSOL-270

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Wrote unit and integration tests.
- [ ] Added relevant natspec and `godoc` comments.
- [ ] Provide a [conventional commit message](https://www.conventionalcommits.org/en/v1.0.0/) to follow the repository standards.
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: e2e/interchaintestv8/solana/client.go
Comment (srdtrk) [inline e2e/interchaintestv8/solana/client.go:1]: Why are all the godocs removed?

File: e2e/interchaintestv8/solana/gmp.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+type AccountState struct {
+	ClientID       string
+	Sender         string
+	Salt           []byte
+	Nonce          uint64
+	CreatedAt      int64
+	LastExecutedAt int64
+	ExecutionCount uint64
+	Bump           uint8
+}
```
Comment (srdtrk) [inline e2e/interchaintestv8/solana/gmp.go]: Can this not be imported from autogenerated code?

File: e2e/interchaintestv8/solana/pda.go
Comment (srdtrk) [inline e2e/interchaintestv8/solana/pda.go:1]: Nice. Surprising that some of these are not autogenerated

File: programs/solana/programs/ics27-gmp/src/constants.rs [hunk 2]
```diff
@@ -37,8 +37,8 @@ pub const ICS27_ENCODING: &str = "application/x-protobuf";
 /// Maximum timeout duration (24 hours in seconds)
 pub const MAX_TIMEOUT_DURATION: i64 = 86400;
 
-/// Minimum timeout duration (1 minute in seconds)
-pub const MIN_TIMEOUT_DURATION: i64 = 60;
+/// Minimum timeout duration (0 seconds - no minimum enforced)
+pub const MIN_TIMEOUT_DURATION: i64 = 0;
```
Comment (srdtrk) [inline programs/solana/programs/ics27-gmp/src/constants.rs]: We should probably do something non-zero. Like 12 seconds

File: programs/solana/programs/ics27-gmp/tests/gmp_tests.rs [hunk 3]
```diff
@@ -179,24 +179,6 @@ fn test_send_call_msg_validation_success() {
...
-        salt: vec![],
-        payload: vec![1, 2, 3],
-        memo: String::new(),
-    };
-
-    assert!(
-        msg.validate(current_time).is_err(),
-        "Timeout too soon should fail"
-    );
-}
```
Comment (srdtrk) [inline programs/solana/programs/ics27-gmp/tests/gmp_tests.rs:198]: You don't need to remove this test even if MIN_TIMEOUT_DURATION is zero. You can test with something less than current_time.

File: e2e/interchaintestv8/solana/pda.go
Comment (mariuszzak) [reply e2e/interchaintestv8/solana/pda.go:1]: Indeed, anchor-go does not generate helpers for PDAs. But I just built a simple generator that parses the IDL files:

https://github.com/cosmos/solidity-ibc-eureka/blob/mariuszzak/e2e-unhappy-paths/e2e/interchaintestv8/solana/generate-pdas/main.go

it generates

https://github.com/cosmos/solidity-ibc-eureka/blob/mariuszzak/e2e-unhappy-paths/e2e/interchaintestv8/solana/pda.go

File: e2e/interchaintestv8/solana/gmp.go (hunk 1 above)
Comment (mariuszzak) [reply e2e/interchaintestv8/solana/gmp.go]: This one is tricky, as it is not included in IDLs. (because we dynamically parse this account in the program's logic, as it is an optional account. The logic creates it when it does not exist.) 

I will try to figure out how to avoid such hardcoding.

File: programs/solana/programs/ics27-gmp/src/constants.rs (hunk 2 above)
Comment (mariuszzak) [reply programs/solana/programs/ics27-gmp/src/constants.rs]: Ok, will update it.

File: programs/solana/programs/ics27-gmp/tests/gmp_tests.rs (hunk 3 above)
Comment (mariuszzak) [reply programs/solana/programs/ics27-gmp/tests/gmp_tests.rs:198]: Ok ok, will restore it.

File: e2e/interchaintestv8/solana/client.go
Comment (mariuszzak) [reply e2e/interchaintestv8/solana/client.go:1]: Dropped by accident during refactoring. Will restore them.

File: e2e/interchaintestv8/solana/gmp.go (hunk 1 above)
Comment (mariuszzak) [reply e2e/interchaintestv8/solana/gmp.go]: Ok, I found a workaround for that. Added a getter instruction, which enforced Anchor to include `AccountState` in IDL (because we use `#[derive(Accounts)]` for this account). 

https://github.com/cosmos/solidity-ibc-eureka/pull/782/commits/4f061e6bbb5c9464b0d90a262ddb3adfb26fc53c
